### ISSUE TITLE
Iterate may not show correctly in apps that manage multiple windows

### ISSUE THREAD

**Issue by brouwers-at-coffeeit** [iteratehq/iterate-ios#78]
**Describe the bug**
The iterate window may not show correctly in apps with multiple windows.

`ContainerWindowDelegate` contains a fallback in `getPresentingViewController` where, if the Iterate window is the keyWindow, it will instead select the first window it can find:

```
// If we didn't find one, get the first non-key window since our window may be key (happens in iOS 14.2)
if window == nil {
   window = UIApplication.shared.windows.first { $0 != self.window }
}
```

In case this `first` window is not the window visible to the user, the Iterate views will not become visible (it's displayed **behind** the visible window). Additionally, because the user cannot see that Iterate window, they cannot close it, so if they navigate to a different part of the app where Iterate would normally be shown correctly, it would fail to show, because it's already showing, i.e. **breaks once, stays broken**.

**To Reproduce**
Steps to reproduce the behavior (may vary):
1. Update to Iterate SDK v1.0.9
2. In your app, open a view in a _new UIWindow_.
3. From within that new view, open any Iterate survey.
4. If by chance Iterate is the key window, it may fallback to previous (main app) window.
5. If this occurs, the Iterate survey views will appear behind the _new UIWindow_.

Note, you may see the 1st part of the survey appear, at which point Iterate becomes the keyWindow, when it tries to show the Feedback form section of a survey, that view will not be shown correctly as described above. And Iterate will stay broken for the remainder of that session.

**Expected behavior**
Instead, I expect Iterate to correctly identify which UIWindow is visible to the user whilst still avoiding itself.
I've created a fork that provides work arounds for the described scenario.

Changes applied on September 28th, 2022, on Iterate v1.0.9
https://github.com/iteratehq/iterate-ios/compare/master...brouwers-at-coffeeit:iterate-ios:master

Pull request:
https://github.com/iteratehq/iterate-ios/pull/79

In short, I added:
1. Support for setting a fallback window tag for Iterate to try and look for before defaulting to using the first window. 
2. Support for setting a tag on Iterate's window, useful for forcing it to be displayed in a certain orientation (landscape) based on circumstances.
3. Support for hiding the container window manually, so Iterate can be reset if it ever reaches a broken state during a session. This mostly eased troubles during testing fixes, but is not actually necessary now our problem has been resolved.

**Device:**
 - Device: iPhone X
 - OS: iOS 16.0

**Comment by msingleton** [comment]
Closing for now with the note in the PR, we'll pick this back up if we get any other customer requests. In the meantime I hope your fork works for you
